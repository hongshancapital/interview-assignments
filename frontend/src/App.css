.App {
  text-align: center;
  position: relative;
}
