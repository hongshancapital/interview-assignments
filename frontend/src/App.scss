.App {
  text-align: center;
  position: relative;
}

.title {
  font-size: 56px;
  font-weight: 600;
}

.text {
  font-size: 28px;
  font-weight: 400;
}
