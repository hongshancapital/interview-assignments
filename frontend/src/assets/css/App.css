.App {
  text-align: center;
  position: relative;
}


/*slides*/
.color {
  mix-blend-mode: difference;
  color: #fff;
}

.slides {
  width: 500px;
  height: 300px;
  overflow: hidden;
  margin: auto;
  position: relative;
}

.slides-item {
  display: flex;
  width: 100%;
  height: 100%;
  transition: transform 1s;
}

.slide {
  position: relative;
  width: 100%;
  background-color: #000;
}

.slide-img {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  object-fit: cover;
}

.slide-title {
  font-size: 24px;
  font-weight: 600;
  white-space: pre-wrap;
  margin: 20px;
}

.slide-text {
  padding: 0 20px;
  margin: 0;
  font-size: 18px;
}

.slides-nav {
  position: absolute;
  bottom: 10px;
  left: 0;
  width: 100%;
  display: flex;
  justify-content: center;
}

.nav-bar {
  width: 30px;
  height: 3px;
  background-color: rgba(0, 0, 0, 0.5);
  margin-right: 5px;
  border-radius: 20px;
  overflow: hidden;
}

.nav-progress {
  background-color: #fff;
  background-blend-mode: difference;
  width: 0;
  height: 100%;
}