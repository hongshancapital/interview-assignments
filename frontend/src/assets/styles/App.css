.App {
  text-align: center;
  position: relative;
}

.title {
  font-size: 56px;
  font-weight: 600;
}

.text {
  font-size: 28px;
  font-weight: 400;
}

.my_carousel_item {
  width: 480px;
  height: 300px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  position: relative;
}

.my_carousel_item2 {
  width: 960px;
  height: 600px;
}

.my_carousel_item3 {
  width: 1440px;
  height: 900px;
}

.my_carousel_item .description {
  position: absolute;
  text-align: center;
  width: 100%;
  top: 36px;
}

.my_carousel_item._white {
  color: #fff;
}

.my_carousel_item h3 {
  font-size: 26px;
}
.my_carousel_item p {
  font-size: 12px;
}

.my_carousel_item2 h3 {
  font-size: 56px;
}
.my_carousel_item2 p {
  font-size: 28px;
}

.my_carousel_item3 h3 {
  font-size: 86px;
}
.my_carousel_item3 p {
  font-size: 43px;
}
