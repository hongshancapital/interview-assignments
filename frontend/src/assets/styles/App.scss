.App {
  text-align: center;
  position: relative;
}

.title {
  font-size: 56px;
  font-weight: 600;
}

.text {
  font-size: 28px;
  font-weight: 400;
}

.my_carousel_item {
  width: 480px;
  height: 300px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  position: relative;
  &.my_carousel_item2 {
    width: 960px;
    height: 600px;
    h3 {
      font-size: 56px;
    }
    p {
      font-size: 28px;
    }
  }
  &.my_carousel_item3 {
    width: 1440px;
    height: 900px;
    h3 {
      font-size: 86px;
    }
    p {
      font-size: 43px;
    }
  }

  &._white {
    color: #fff;
  }
  .description {
    position: absolute;
    text-align: center;
    width: 100%;
    top: 36px;
  }

  h3 {
    font-size: 26px;
  }
  p {
    font-size: 12px;
  }
}
