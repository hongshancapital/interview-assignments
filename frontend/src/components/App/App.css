.App {
    width: 100%;
    height: 100%;
    position: relative;
}

.full-width {
    width: 100%;
}

.center {
    position: absolute;
    width: 100%;
    text-align: center;
    top: 20%;
}

.color--white {
    color: white;
}
