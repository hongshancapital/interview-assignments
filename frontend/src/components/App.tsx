import React from "react";
import Carousel from "./Carousel";
import { SLIDES_DATA, DELAY_TIME } from "../common/constants";
import "../assets/css/App.css";

function App() {
  return (
  	<div className="App">
  		<Carousel slides={SLIDES_DATA} delay={DELAY_TIME} />
  	</div>
  );
}

export default App;
