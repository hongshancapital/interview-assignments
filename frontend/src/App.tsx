import './App.css';
import { Carousel } from './components/Carousel';
import { APP_CAROUSEL_ITEMS, CarouselItem } from './components/CarouselItem';

function App() {
  return (
    <div className="App">
      <Carousel>
        {APP_CAROUSEL_ITEMS.map((item) => (
          <CarouselItem key={item.title} {...item} />
        ))}
      </Carousel>
    </div>
  );
}

export default App;
