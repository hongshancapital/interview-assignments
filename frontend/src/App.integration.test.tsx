import { act, render, screen } from '@testing-library/react';
import App from './App';

describe('App', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  describe('Carousel', () => {
    it('renders Carousel with CarouselItems', () => {
      const { container } = render(<App />);

      expect(container).toMatchSnapshot();
    });

    it('displays correct index of CarouselItem and sets correct indicator style after default duration', () => {
      render(<App />);

      expect(
        screen.queryByRole('heading', { name: 'Tablet' })
      ).not.toBeInTheDocument();
      expect(screen.getAllByLabelText('indicator progress')[1]).toHaveStyle({
        opacity: '0',
        animation: 'none',
      });

      act(() => {
        jest.advanceTimersByTime(3000);
      });

      expect(screen.getByRole('heading', { name: 'Tablet' })).toBeVisible();
      expect(screen.getAllByLabelText('indicator progress')[1]).toHaveStyle({
        opacity: '1',
        animation: 'progress 3000ms linear',
      });
    });
  });
});
