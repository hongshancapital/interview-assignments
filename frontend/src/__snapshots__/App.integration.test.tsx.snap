// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Carousel renders Carousel with CarouselItems 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="carousel"
    >
      <ul
        aria-label="carousel items"
        class="carouselItems"
        style="width: 300%; transform: translateX(calc(-1 * 100% / 3 * 0));"
      >
        <li
          aria-hidden="false"
          aria-label="carousel item"
          class="carouselItemWrapper"
        >
          <div
            class="carouselItem carouselItemIphone"
          >
            <h2
              class="carouselItemTitle"
            >
              xPhone
            </h2>
            <h3
              class="carouselItemSubTitle"
            >
              Lots to love.Less to spend.
Starting at $399.
            </h3>
            <img
              alt="xPhone"
              class="carouselItemImage"
              src="iphone.png"
            />
          </div>
        </li>
        <li
          aria-hidden="true"
          aria-label="carousel item"
          class="carouselItemWrapper"
        >
          <div
            class="carouselItem carouselItemTablet"
          >
            <h2
              class="carouselItemTitle"
            >
              Tablet
            </h2>
            <h3
              class="carouselItemSubTitle"
            >
              Just the right amount of everything.
            </h3>
            <img
              alt="Tablet"
              class="carouselItemImage"
              src="tablet.png"
            />
          </div>
        </li>
        <li
          aria-hidden="true"
          aria-label="carousel item"
          class="carouselItemWrapper"
        >
          <div
            class="carouselItem carouselItemAirPod"
          >
            <h2
              class="carouselItemTitle"
            >
              Buy a Tablet or xPhone for college.
Get airPods.
            </h2>
            <h3
              class="carouselItemSubTitle"
            />
            <img
              alt="Buy a Tablet or xPhone for college.
Get airPods."
              class="carouselItemImage"
              src="airpods.png"
            />
          </div>
        </li>
      </ul>
      <ul
        class="indicators"
      >
        <li
          aria-label="indicator"
          class="indicator"
          role="progressbar"
        >
          <span
            aria-label="indicator progress"
            class="indicatorProgress"
            style="opacity: 1; animation: progress 3000ms linear;"
          />
        </li>
        <li
          aria-label="indicator"
          class="indicator"
          role="progressbar"
        >
          <span
            aria-label="indicator progress"
            class="indicatorProgress"
            style="opacity: 0; animation: none;"
          />
        </li>
        <li
          aria-label="indicator"
          class="indicator"
          role="progressbar"
        >
          <span
            aria-label="indicator progress"
            class="indicatorProgress"
            style="opacity: 0; animation: none;"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
